src/main/java/eu/bbmri/eric/csit/service/negotiator/api/dto/post/PostCreateDTO.java

```diff
@@ -0,0 +1,34 @@
+package eu.bbmri.eric.csit.service.negotiator.api.dto.post;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import eu.bbmri.eric.csit.service.negotiator.database.model.Attachment;
+import java.util.Set;
+import javax.validation.Valid;
+import javax.validation.constraints.NotEmpty;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@JsonInclude(value = JsonInclude.Include.NON_NULL)
+public class PostCreateDTO {
+
+  Set<Attachment> attachments;
+  @Valid
+  @NotEmpty
+  private String negotiationId;
+  @Valid
+  @NotEmpty
+  private String text;
+  @Valid
+  @NotEmpty
+  private String resourceId;
```

Comment:
The resource Id could be empty for a public post. We still do not have a feature for private posts so I would remove it for now

Reply:
How about leaving it but allowing it to be empty? To remove it  would have  a big impact as in the current flow at least 1 resource is specified when creating a post, and in the feature we would need it

Reply:
Okay then just allow it to be empty